File: UI/src/app/shared/models/company.model.ts
import { UserModel } from "./user.model";

export class CompanyModel {

  id: number = 0;
  userId: number = 0;

  companyName: string = '';
  website?: string;
  description?: string;
  industry?: string;
  location?: string;

  hrName?: string;
  contactEmail?: string;
  contactPhone?: string;

  foundedYear?: number;
  companySize?: number;

  logoUrl?: string;

  createdAt: Date = new Date();

  // Navigation
  user?: UserModel;

}

File: API/Model/CompanyModel.cs
namespace API.Model
{
    public class CompanyModel
    {
        public long Id { get; set; }
        public long UserId { get; set; }
        public string CompanyName { get; set; } = string.Empty;
        public string? Website { get; set; }
        public string? Description { get; set; }
        public string? Industry { get; set; }
        public string? Location { get; set; }
        public string? HRName { get; set; }
        public string? ContactEmail { get; set; }
        public string? ContactPhone { get; set; }
        public int? FoundedYear { get; set; }
        public int? CompanySize { get; set; }
        public string? LogoUrl { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        // Navigation
        public List<UserModel>? Users { get; set; }
    }
    public class CompanyCreateDTO
    {
        public IFormFile? CompanyLogo { get; set; }
        public string? Data { get; set; }
    }
}
